connector/keystone/keystone.go

```diff
@@ -0,0 +1,158 @@
+// Package keystone provides authentication strategy using Keystone.
+package keystone
+
+import (
+	"context"
+	"fmt"
+	"github.com/dexidp/dex/connector"
+	"github.com/sirupsen/logrus"
+	"encoding/json"
+	"net/http"
+	"bytes"
+	"io/ioutil"
+)
+
+var (
+	_ connector.PasswordConnector = &keystoneConnector{}
+  	_ connector.RefreshConnector = &keystoneConnector{}
+)
+
+// Open returns an authentication strategy using Keystone.
+func (c *Config) Open(id string, logger logrus.FieldLogger) (connector.Connector, error) {
+	return &keystoneConnector{c.Domain, c.KeystoneHost,
+	c.KeystoneUsername, c.KeystonePassword, logger}, nil
+}
+
+func (p keystoneConnector) Close() error { return nil }
+
+func (p keystoneConnector) Login(ctx context.Context, s connector.Scopes, username, password string) (
+		identity connector.Identity, validPassword bool, err error) {
+	response, err := p.getTokenResponse(username, password)
+
+	// Providing wrong password or wrong keystone URI throws error
+	if err == nil && response.StatusCode == 201 {
+    	token := response.Header["X-Subject-Token"][0]
+		data, _ := ioutil.ReadAll(response.Body)
```

Comment:
Let's please check all errors (same applies to other calls in this PR) -- it's fine to skip an error check or two in the test files, but let's please not do that here.